R/model_income_tax.R

```diff
@@ -168,6 +172,14 @@ model_income_tax <- function(sample_file,
          paste0(absent_cols, collapse = "\n\t"), ".\n")
   }
   
+  #cg adjustment
+  if (!is.null(cg_discount_rate)) {
+    sample_file[, Taxable_Income := ifelse(Net_CG_amt==Tot_CY_CG_amt & Tot_CY_CG_amt > 0, #assume ineligible if Net_CG_amt==Tot_CY_CG_amt
+                                           Taxable_Income,
+                                           pmaxC(Taxable_Income + Net_CG_amt * (1 - 2 * cg_discount_rate),
```

Comment:
Just to document how I got this:
Taxable_Income <- taxable_income  - Net_CG_amt + model_CG_amt
model_cg_amt <- Net_CG_amt * 2 * (1 - cg_discount_rate)

Taxable_Income <- taxable_income  - Net_CG_amt + Net_CG_amt * 2 * (1 - cg_discount_rate)
                          <- taxable_income  + Net_CG_amt * (2 * (1 - cg_discount_rate) - 1)
                          <- taxable_income  + Net_CG_amt * (1 - 2* cg_discount_rate)

Reply:
That's good.

Never use `ifelse` though -- it's slow and can be a bit dangerous. Use `if_else`.

Compare:

```r
sample_file[, Taxable_Income := ifelse(is.null(cg_discount_rate), Taxable_Income, Taxable_Income + Net_CG_amt * cgt_discount_rate)]
```

with

```r
sample_file[, Taxable_Income := if_else(is.null(cg_discount_rate), Taxable_Income, Taxable_Income + Net_CG_amt * cgt_discount_rate)]
```

You can also see what `microbenchmark` reveals about the two operations.